src/tinygraph.c

```diff
@@ -31,10 +31,13 @@ TINY_GRAPH *TinyGraphConnect(TINY_GRAPH *G, int i, int j)
 {
     if(TinyGraphAreConnected(G, i, j))
 	return G;
+    if(i==j) assert(G->selfLoops);
     TSetAdd(G->A[i], j);
-    if(!G->directed) TSetAdd(G->A[j],i);
     ++G->degree[i];
-    if(i==j) assert(G->selfLoops);
+    if(!G->directed&&i!=j){
```

Comment:
To make the logic a bit more clear, move the if(i==j) assert(G->self) here, and then add an "else if(!directed)". Same with Disconnect.